controllers/nutanixmachine_controller.go

```diff
@@ -327,81 +311,109 @@ func (r *NutanixMachineReconciler) reconcileDelete(rctx *nctx.MachineContext) (r
 		log.Error(errorMsg, "failed to delete VM")
 		return reconcile.Result{}, errorMsg
 	}
+
 	// Check if VMUUID is absent
 	if vmUUID == "" {
 		log.Info(fmt.Sprintf("VMUUID was not found in spec for VM %s. Skipping delete", vmName))
-	} else {
-		// Search for VM by UUID
-		vm, err := FindVMByUUID(ctx, v3Client, vmUUID)
-		// Error while finding VM
+		log.Info(fmt.Sprintf("Removing finalizers for VM %s during delete reconciliation", vmName))
+		ctrlutil.RemoveFinalizer(rctx.NutanixMachine, infrav1.NutanixMachineFinalizer)
+		return reconcile.Result{}, nil
+	}
+
+	// Search for VM by UUID
+	vm, err := FindVMByUUID(ctx, v3Client, vmUUID)
```

Comment:
Its hard to read the diff from existing code, did any existing code change thats not related to the VG detachment?

Reply:
I'd recommend reading the diff with the setting on whitespace removal separated. This PR tries to keep cyclomatic complexity in check by removing some outer loops in place.